mitosheet/mitosheet/step_performers/graph_steps/plotly_express_graphs.py

```diff
@@ -189,27 +195,65 @@ def graph_creation_code(
 
 
 def graph_styling(
-    fig: go.Figure, graph_type: str, column_headers: List[ColumnHeader], filtered: bool
+    fig: go.Figure, graph_type: str, column_headers: List[ColumnHeader], filtered: bool, graph_styling_params: Dict[str, Any]
 ) -> go.Figure:
     """
     Styles the Plotly express graph figure
     """
 
     # Create the parameters that we use to style the graph
-    graph_title = get_graph_title(column_headers, [], filtered, graph_type)
+    if graph_styling_params['title']['visible']:
+        graph_title = graph_styling_params['title']['title'] if 'title' in graph_styling_params['title'].keys() else get_graph_title(column_headers, [], filtered, graph_type)
+    else:
+        graph_title = None
+
+    # If the x axis title is visible, then either dispaly the user's custom title or Ploty's default title.
+    # Otherwise, get rid of the title
+    if graph_styling_params['xaxis']['visible']:
+        use_custom_xaxis_title = 'title' in graph_styling_params['xaxis'].keys()
+        # Only apply the xaxis_title if it is set because if we set it to None, then we don't get ploty's default values
+        if use_custom_xaxis_title:
+            fig.update_layout(
+                xaxis_title = graph_styling_params['xaxis']['title']
+            )
+    else: 
+        fig.update_layout(
+            xaxis_title = None
+        )
+
+    # If the y axis title is visible, then either dispaly the user's custom title or Ploty's default title.
+    # Otherwise, get rid of the title
+    if graph_styling_params['yaxis']['visible']:
+        use_custom_yaxis_title = 'title' in graph_styling_params['yaxis'].keys()
+        # Only apply the xaxis_title if it is set because if we set it to None, then we don't get ploty's default values
+        if use_custom_yaxis_title:
+            fig.update_layout(
```

Comment:
Building a parameters dictionary, as you do with graph creation, as clearly the best bet here! I think it will lead to much more maintainable and clean code over time!